File: client/src/app/modules/services/dto/project-delete-dto.ts
import {ProjectTableRowDto} from "./project-table-row-dto";

export class ProjectDeleteDto {

    id: number;

    projectNumber: number;

    status: string;

    version: number;

    constructor(projectTableRowDto: ProjectTableRowDto) {
        this.id = projectTableRowDto.id;
        this.projectNumber = projectTableRowDto.projectNumber;
        this.status = projectTableRowDto.status.id;
        this.version = projectTableRowDto.version;
    }

}

File: server/src/main/java/vn/elca/training/tdd/dto/ProjectDeleteDto.java
package vn.elca.training.tdd.dto;

import lombok.Data;
import vn.elca.training.tdd.dom.Project;
import vn.elca.training.tdd.dom.enums.ProjectStatus;

@Data
public class ProjectDeleteDto extends AbstractVersionDto {
    
    private long id;
    
    private int projectNumber;
    
    private ProjectStatus status;
    
    public Project getProject() {
        Project project = new Project();
        
        project.setId(this.id);
        project.setProjectNumber(this.projectNumber);
        project.setStatus(this.status);
        project.setVersion(this.version);
        
        return project;
    }
    
}
